src/filters.rs

```diff
@@ -271,47 +277,191 @@ pub fn last(input: &Value, _args: &[Value]) -> FilterResult {
-                    for string in strings_to_join {
-                        result.push_str(join_string);
-                        result.push_str(&string);
-                    }
-                    Ok(Str(result))
-                }
-                _ => Err(InvalidArgument(0, "expected String argument as join".to_owned())),
+        Str(ref x) => {
+            match args.first() {
+                Some(&Str(ref a)) => Ok(Str(x.replace(a, ""))),
+                _ => Err(InvalidArgument(0, "Str expected".to_owned())),
             }
         }
-        _ => Err(InvalidType("Array of Strings expected".to_owned())),
+        _ => Err(InvalidType("String expected".to_owned())),
+    }
+}
+
+pub fn remove_first(input: &Value, args: &[Value]) -> FilterResult {
+    match *input {
+        Str(ref x) => {
+            match args.first() {
+                Some(&Str(ref a)) => Ok(Str(x.splitn(2, a).collect())),
+                _ => Err(InvalidArgument(0, "Str expected".to_owned())),
+            }
+        }
+        _ => Err(InvalidType("String expected".to_owned())),
+    }
+}
+
+pub fn replace(input: &Value, args: &[Value]) -> FilterResult {
+    try!(check_args_len(args, 2));
+    match *input {
+        Str(ref x) => {
+            let arg1 = match args[0] {
+                Str(ref a) => a,
+                _ => return Err(InvalidArgument(0, "Str expected".to_owned())),
+            };
+            let arg2 = match args[1] {
+                Str(ref a) => a,
+                _ => return Err(InvalidArgument(1, "Str expected".to_owned())),
+            };
+            Ok(Str(x.replace(arg1, arg2)))
+        }
+        _ => Err(InvalidType("String expected".to_owned())),
+    }
+}
+
+pub fn replace_first(input: &Value, args: &[Value]) -> FilterResult {
+    if args.len() != 2 {
+        return Err(InvalidArgumentCount(format!("expected 2, {} given", args.len())));
+    }
+    match *input {
+        Str(ref x) => {
+            let search = match args[0] {
+                Str(ref a) => a,
+                _ => return Err(InvalidArgument(0, "Str expected".to_owned())),
+            };
+            let replace = match args[1] {
+                Str(ref a) => a,
+                _ => return Err(InvalidArgument(1, "Str expected".to_owned())),
+            };
+            let tokens: Vec<&str> = x.splitn(2, search).collect();
+            if tokens.len() == 2 {
+                let result = tokens[0].to_string() + replace + tokens[1];
+                Ok(Str(result))
+            } else {
+                Ok(Str(x.to_string()))
+            }
+        }
+        _ => Err(InvalidType("String expected".to_owned())),
+    }
+}
+
+/// Reverses the order of the items in an array. `reverse` cannot `reverse` a string.
+pub fn reverse(input: &Value, args: &[Value]) -> FilterResult {
+    try!(check_args_len(args, 0));
+    match *input {
+        Value::Array(ref array) => {
+            let mut reversed = array.clone();
```

Comment:
Note to self: Look if we can reduce allocation in filters (by modifying `output.rs`) before 1.0